File: JavaScript/프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    var stack = [];
    var count = 0;
    for (let i=0; i<moves.length; i++) {
        let pos = moves[i]-1;
        for (let j=0; j<board.length; j++){
            if (board[j][pos] != 0) {
                stack.push(board[j][pos]);
                board[j][pos] = 0;
                break;
            }
        }
        if (stack.length >= 2 && stack[stack.length - 1] == stack[stack.length - 2]) {
            stack.pop();
            stack.pop();
            count += 2;
        }
    }
    return count;
}

File: Python3/프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    count = 0
    stack = []
    for pos in moves:
        pos -= 1
        for row in board:
            if row[pos] != 0:
                stack.append(row[pos])
                row[pos] = 0
                break
        if len(stack) >= 2 and stack[-1] == stack[-2]:
            stack.pop()
            stack.pop()
            count += 2
            
    return count